src/main/java/org/springframework/data/redis/listener/RedisMessageListenerContainer.java

```diff
@@ -759,6 +779,7 @@ public void run() {
 					}
 				}
 			} catch (Throwable t) {
+				cancelPatternSubscriptionTask();
```

Comment:
since this whole code block is rerun on error, I cancel the task here so that everything is clean for the next run. Since it's synchronized by `localMonitor`, everything should be ordered.